### PR TITLE
Add new APIs to Blob Swagger

### PR DESCRIPTION
**Pull request by PaulWells**
Adds PageBlob_ClearPages, Blobs_Resize and Blobs_UpdateSequenceNumber.

### CODE REVIEW

File: Specs/blob/blob-storage-2017-07-29.json [hunk 1]
```diff
@@ -2059,10 +2059,10 @@
             "$ref": "#/parameters/IfNoneMatch"
           },
           {
-            "$ref": "#/parameters/BlobContentLength"
+            "$ref": "#/parameters/BlobContentLengthOptional"
```
Comment (sakirsch-msft) [inline Specs/blob/blob-storage-2017-07-29.json]: Should we be making many of the other parameters optional for the other APIs?

Comment (sakirsch-msft) [reply Specs/blob/blob-storage-2017-07-29.json]: Are there other parameters we should be making required then? For example some of the lease operations would require lease ids.

Comment (PaulWells) [reply Specs/blob/blob-storage-2017-07-29.json]: Yes, you're right. A couple of the lease API parameters we should make optional/remove. Let's create another task for that

File: Specs/blob/blob-storage-2017-07-29.json [hunk 2]
```diff
@@ -2430,13 +2430,224 @@
             "$ref": "#/parameters/BlobContentDisposition"
           },
           {
-            "$ref": "#/parameters/BlobContentLength"
+            "$ref": "#/parameters/BlobSequenceNumberOptional"
```
Comment (sakirsch-msft) [inline Specs/blob/blob-storage-2017-07-29.json]: Feels like this should be part of a separate API under Page Blobs.

File: Specs/blob/blob-storage-2017-07-29.json [hunk 3]
```diff
@@ -2430,13 +2430,224 @@
...
+          ]
+        }
+      ]
+    },
+    "/{containerName}/{blob}?comp=properties&Resize": {
+      "put": {
+        "tags": [
+          "blob"
+        ],
+        "operationId": "Blobs_Resize",
```
Comment (sakirsch-msft) [inline Specs/blob/blob-storage-2017-07-29.json]: page blob only?

File: Specs/blob/blob-storage-2017-07-29.json [hunk 4]
```diff
@@ -2430,13 +2430,224 @@
...
+          ]
+        }
+      ]
+    },
+    "/{containerName}/{blob}?comp=properties&UpdateSequenceNumber": {
+      "put": {
+        "tags": [
+          "blob"
+        ],
+        "operationId": "Blobs_UpdateSequenceNumber",
```
Comment (sakirsch-msft) [inline Specs/blob/blob-storage-2017-07-29.json]: page blob only?

File: Specs/blob/blob-storage-2017-07-29.json [hunk 5]
```diff
@@ -3711,6 +3922,122 @@
...
+            "$ref": "#/parameters/Timeout"
+          },
+          {
+            "$ref": "#/parameters/Range"
+          },
+          {
+            "$ref": "#/parameters/PageWriteClear"
+          },
+          {
+            "$ref": "#/parameters/LeaseId"
```
Comment (sakirsch-msft) [inline Specs/blob/blob-storage-2017-07-29.json]: Note to check that when the other PR gets merged that all of these are updated as well

File: Specs/blob/blob-storage-2017-07-29.json [hunk 6]
```diff
@@ -3711,6 +3922,122 @@
...
+          "name": "comp",
+          "in": "query",
+          "required": true,
+          "type": "string",
+          "enum": [
+            "page"
+          ]
+        }
+      ]
+    },
```
Comment (sakirsch-msft) [inline Specs/blob/blob-storage-2017-07-29.json]: Are we also doing Get Page Ranges Diff?

Comment (sakirsch-msft) [inline Specs/blob/blob-storage-2017-07-29.json]: Are we also doing List Blob Hierarchy Segment? / Flat?

Comment (PaulWells) [reply Specs/blob/blob-storage-2017-07-29.json]: added List Blob Hierarchy Segment because it was easier to add then think about whether to add it

**Pull request merged**